### ISSUE TITLE
Deno no longer working on darwin due to tinycc being broken

### ISSUE THREAD

**Issue by khaled** [NixOS/nixpkgs#181982]
### Describe the bug
Deno no longer working on darwin due to tinycc being broken.

### Steps To Reproduce
```
❯ nix run nixpkgs#deno
```
`error: Package ‘tcc-0.9.27+date=2022-01-11’ in /nix/store/4la4fc5wfn4p0784i12h0a046yqzln16-source/pkgs/development/compilers/tinycc/default.nix:65 is marked as broken, refusing to evaluate.
`

### Expected behavior
It works.

### Additional context
Looks like this happened with the move to Deno 1.23.4: https://github.com/NixOS/nixpkgs/commit/18348a5e4d1e4654f2bc7e6e5c9746ca56c1dd5e

Issue related to tinycc being broken: https://github.com/NixOS/nixpkgs/issues/100749

### Notify maintainers
deno: @06kellyjac
tinycc: @joachifm @AndersonTorres

### Metadata

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.5.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(xxyy): `"darwin"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by 06kellyjac** [comment]
Thanks for raising this. I shouldn't have missed this.

I'll try fix tinycc and if we can't Mac might be able to fall back to the same non-tinycc method (even if it has some performance penalties) until they migrate to asm

**Comment by khaled** [comment]
Thanks for being on this @06kellyjac!

**Comment by 06kellyjac** [comment]
@khaled if you could try out #182311 that'd be helpful

**Comment by khaled** [comment]
@06kellyjac i'm on it!

**Comment by khaled** [comment]
@06kellyjac `nixpkgs-review pr 182311` succeeded for me on x86_64-darwin

**Comment by 06kellyjac** [comment]
Nice, it worked for me in my x86_64-darwin vm but it's always good to get it checked on real hardware. Would be nice to have someone try it on aarch64-darwin

As a note if you want to do a "full" test of the FFI features (the one that requires tcc now)

```
# clone the deno repo somewhere
git clone https://github.com/denoland/deno /tmp/deno
cd /tmp/deno

# copy deno to the correct folder - the tests get upset if the deno binary is not in the right folder so we need to make it and copy it over from our nixpkgs build
mkdir -p ./target/result
cp /location/of/the/nixpkgs/result/bin/deno ./target/result/

# then run the ffi benchmarks
target/release/deno bench --allow-ffi --allow-read --unstable ./test_ffi/tests/bench.js

**Comment by cideM** [comment]
Ran it on my aarch64-darwin machine:
```shell
2 packages marked as broken and skipped:
nrpl tinycc

1 package built:
deno
[nix-shell:~/.cache/nixpkgs-review/pr-182311]$ deno
Deno 1.25.0
exit using ctrl+d or close()
> 2 + 2
4
```

**Comment by 06kellyjac** [comment]
fantastic thanks @cideM

**Comment by khaled** [comment]
@06kellyjac also worked for me on aarch64 (i used an EC2 m1 macos instance).  Unfortunately running the FFI tests requires a `libtest_ffy.dylib` to be in the `target/release` directory, which I noticed is output when running `cargo build` outside of nix.  However, on aarch64, I can't get a direct `cargo build` to succeed...

**Comment by 06kellyjac** [comment]
sorry I missed a step there try `cargo build -p test_ffi` if you still have the instance

**Comment by khaled** [comment]
@06kellyjac cool, that worked.  the benchmarks ran without error...

**Comment by aiotter** [comment]
Reopening as #182311 does not resolve this issue

**Comment by 06kellyjac** [comment]
That PR **does** resolve the issue, see the comments towards the bottom to pull in the fix early

The unstable channel hasn't progressed for [6 days](https://status.nixos.org/grafana/d/LhIq8iLWk/channel-updates?orgId=1&refresh=1m&from=now-7d&to=now) and not in scope for this issue

I will open a backport ticket for 22.05 though

**Comment by SuperSandro2000** [comment]
Yes, this is confirmed by ofborg.

**Comment by aiotter** [comment]
OMG Sorry that I didn't follow the PR correctly. Not only I missed the part you fixed on the derivation of deno, but I saw the tinycc's derivation flagged as `meta.broken = stdenv.isDarwin;` and mistook that it is the same on deno's.
The old message of the beginning of the PR ("tinycc darwin issues were a pain, there's no point holding up this update since it's already broken") also made me confused and gave an idea that this issue was closed by mistake.

Sorry for confusing everyone here, and thank you very much for fixing the problem!
